src/NzbDrone.Core/Download/CompletedDownloadService.cs

```diff
@@ -133,9 +149,27 @@ public void Import(TrackedDownload trackedDownload)
                 return;
             }
 
+            var grabbedHistories = _historyService.FindByDownloadId(trackedDownload.DownloadItem.DownloadId).Where(h => h.EventType == EpisodeHistoryEventType.Grabbed).ToList();
+
+            if (grabbedHistories.Count == 0)
+            {
+                grabbedHistories = new List<EpisodeHistory> { new EpisodeHistory() };
+            }
+
             if (trackedDownload.RemoteEpisode == null)
             {
                 trackedDownload.Warn("Unable to parse download, automatic import is not possible.");
+
+                if (!trackedDownload.HasNotifiedManualInteractionRequired)
```

Comment:
I think this can all be pulled out to a separate method that can get the history (if not provided), check whether it's been notified before and send the notification. This way we don't need to look up the history if it's never going to be used and don't have to duplicate this same block of code whether we are checking whether to send a notification.

Reply:
split out the notification decision into a separate method